### PR TITLE
Use the `crypto_base` version for ANSSI

### PR DESCRIPTION
**Pull request by tbrezot**

### CODE REVIEW

**Review by Manuthor — APPROVED** [review]
try this locally and probably take some of those clippy-nursery proposals (probably all IMHO).
```
cargo +nightly clippy --all-targets --all-features --workspace --fix -Z unstable-options --allow-dirty --allow-staged --no-deps --broken-code -- --no-deps -D warnings -W clippy::nursery
```

otherwise all good to me

**Review by ThibsG — COMMENTED** [review]
Nice work thanks!

Some NITs but LGTM :wink:

File: src/interfaces/statics.rs [hunk 1]
```diff
@@ -51,8 +55,15 @@ pub fn encrypt_hybrid_header<DEM: Dem>(
         cover_crypt.generate_symmetric_key(policy, public_key, attributes, DEM::Key::LENGTH)?;
     let encapsulation = encapsulation.try_to_bytes()?;
 
-    // create header
-    let mut header_bytes = Vec::new();
+    // Allocate enough space dot the header bytes
```
[the remark this answers is not in the document]
Comment (tbrezot) [reply src/interfaces/statics.rs:58]: Ahah yes there is !

File: src/cover_crypt_core.rs [hunk 2]
```diff
@@ -189,20 +189,21 @@ pub struct UserPrivateKey {
 }
 
 impl UserPrivateKey {
-    /// Return a reference to the `x` hashmap.
+    /// Return a reference to a hashmap containing the scalars `x_i` for
+    /// each subset `S_i` the user has been given the rights.
     pub fn map(&self) -> &HashMap<Partition, X25519PrivateKey> {
```
[the remark this answers is not in the document]
Comment (tbrezot) [reply src/cover_crypt_core.rs:188]: I made the hashmaps `pub(crate)` for the `UserPrivateKey`, `MasterSecretKey` and the `PublicKey`.

File: src/cover_crypt_core.rs [hunk 3]
```diff
@@ -107,21 +106,22 @@ pub struct MasterPrivateKey {
...
-        serializer.write_array(self.s.to_bytes())?;
-        serializer.write_array((self.x.len() as u32).to_be_bytes().to_vec())?;
+        serializer.write_array(self.u.as_bytes())?;
+        serializer.write_array(self.v.as_bytes())?;
+        serializer.write_array(self.s.as_bytes())?;
+        serializer.write_array(&(self.x.len() as u32).to_be_bytes())?;
         for (partition, x_i) in &self.x {
-            serializer.write_array(partition.into())?;
-            serializer.write_array(x_i.to_bytes())?;
+            serializer.write_array(&Vec::from(partition))?;
```
[the remark this answers is not in the document]
Comment (tbrezot) [reply src/cover_crypt_core.rs:117]: I have implemented `Deref` for `Partition`. This allows to call `serializer.write_array(partition)`.

**Pull request merged**